### DISCUSSION TITLE
Apex charts - more control

### DISCUSSION THREAD

**Discussion by andrewleith** [Budibase/budibase#4275] [Ideas]
Is there any intention to expose the underlying chart components to allow users to use more of the apex chart features (i.e., zoom, etc.)? Or is there a roadmap for more options within the designer UI?

Thanks for this platform btw,  you guys have done an excellent job!

Andrew

**Comment by mjashanks** [comment]
Thanks! 

We have nothing official on the roadmap for this right now, but charts definitely need some work

It's great to get your ideas on this - thank you, keep 'em coming!

**Reply by andrewleith** [reply]
No worries, thanks for reply.  

Off the top of my head I would think you could start with adding options for:

- zoom (bool)
- legend (ability to rename the categories)

A low effort way to increase charting flexibility could be to provide a way for power users to manually edit the json config you pass to apex charts.  Not sure if that is an avenue you guys want to do though.

Thanks again for the excellent platform.
Andrew

**Comment by micebrain** [comment]
What I am missing in the chart is the ability to have more than one series. Chart is useful for showing comparisons but with only one data series it's quite limiting. 
Would be nice to see an improvement in this :)

**Reply by andrewleith** [reply]
I've got multiple series on my area chart, when I get home I'll explain what I did!

**Reply by andrewleith** [reply]
Actually I was thinking of the chart I needed to do groupings in - for that I created a view of my table that aggregated and grouped the data, and then I charted that.  

For the multiple series, it was available right from the area chart options.  My chart looks like this: 
![image](https://user-images.githubusercontent.com/823749/152024050-72435527-1de7-4435-b343-a411368cda74.png)

The *Data Cols.* property in the editor is actually a multi-select, so you can plot more than one piece of data.  Does that help?

**Reply by micebrain** [reply]
Thank you @andrewleith ! I didn't realise that Data Cols is a multi-select and that's because I was trying with the Donut chart which only allow one series. Testing with Area chart I can do the multi-select.
